code/modules/reagents/reagents/medical.dm

```diff
@@ -774,39 +768,56 @@
 	name = "Quick Clot Plus"
 	description = "A chemical designed to quickly and painfully remove internal bleeding by encouraging coagulation. Should not be self-administered."
 	color = "#CC00FF"
-	overdose_threshold = REAGENTS_OVERDOSE/5 //Was 4, now 6 //Now 15 //Now 6 again, yay oldQC
-	overdose_crit_threshold = REAGENTS_OVERDOSE_CRITICAL/5 //10u
+	overdose_threshold = REAGENTS_OVERDOSE/5 //6u
+	overdose_crit_threshold = REAGENTS_OVERDOSE_CRITICAL/5 //12u
 	scannable = TRUE
 	custom_metabolism = REAGENTS_METABOLISM * 2.5
+	///The IB wound this dose of QCP will cure, if it lasts long enough
+	var/datum/wound/internal_bleeding/target_IB
 
 /datum/reagent/medicine/quickclotplus/on_mob_add(mob/living/L, metabolism)
-	if(TIMER_COOLDOWN_CHECK(L, name))
+	if(!ishuman(L))
 		return
-	L.adjustCloneLoss(5*effect_str)
+	var/mob/living/carbon/human/body = L
+	for(var/datum/limb/possible_limb in body.limbs)
+		for(var/datum/wound/internal_bleeding/possible_IB in possible_limb.wounds)
+			target_IB = possible_IB
+			RegisterSignal(target_IB, COMSIG_PARENT_QDELETING, .proc/clear_wound)
+			break
+		if(target_IB)
+			break
+	if(target_IB)
+		to_chat(body, span_highdanger("The deep ache in your [target_IB.parent_limb.display_name] erupts into searing pain!"))
 
 /datum/reagent/medicine/quickclotplus/on_mob_delete(mob/living/L, metabolism)
-	TIMER_COOLDOWN_START(L, name, 30 SECONDS)
+	if(target_IB)
+		to_chat(L, span_warning("The searing pain in your [target_IB.parent_limb.display_name] returns to a dull ache..."))
+		UnregisterSignal(target_IB, COMSIG_PARENT_QDELETING)
+		target_IB = null
 
 /datum/reagent/medicine/quickclotplus/on_mob_life(mob/living/L, metabolism)
-	var/mob/living/carbon/human/H = L
-	for(var/datum/limb/X in H.limbs)
-		for(var/datum/wound/internal_bleeding/W in X.wounds)
-			W.damage = max(0, W.damage - (2.5*effect_str))
 	L.reagents.add_reagent(/datum/reagent/toxin,5)
 	L.reagent_shock_modifier -= PAIN_REDUCTION_VERY_HEAVY
 	L.adjustStaminaLoss(15*effect_str)
+	if(current_cycle == 9 && target_IB) //Minimum 5u per dose, and make sure nothing else has cleared it up in the meantime.
```

Comment:
9 is a magic number

Reply:
A lot of reagent code has hardcoded cycle checks, where would you prefer it?